src/main.cpp

```diff
@@ -2871,9 +2871,18 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
     }
     blockundo.old_sprout_tree_root = old_sprout_tree_root;
 
-    // If Sapling is active, block.hashFinalSaplingRoot must be the
-    // same as the root of the Sapling tree
-    if (chainparams.GetConsensus().NetworkUpgradeActive(pindex->nHeight, Consensus::UPGRADE_SAPLING)) {
+    // If Heartwood is active, block.hashHistoryRoot must be the same as previous
+    // block history root (which is not yet updated)
+    if (chainparams.GetConsensus().NetworkUpgradeActive(pindex->nHeight, Consensus::UPGRADE_HEARTWOOD)) {
```

Comment:
this probably should be here as temporary versioning while upgrade is not activated, but I might be wrong